File: SDKs/drchrono.com/v4 (Hunt Valley)/python/src/sdk/models/operations/care_plans_list.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import security as shared_security
from ..shared import careplan as shared_careplan


@dataclasses.dataclass
class CarePlansListQueryParams:
    cursor: Optional[str] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'cursor', 'style': 'form', 'explode': True }})
    doctor: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'doctor', 'style': 'form', 'explode': True }})
    page_size: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'page_size', 'style': 'form', 'explode': True }})
    patient: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'patient', 'style': 'form', 'explode': True }})
    plan_type: Optional[int] = dataclasses.field(default=None, metadata={'query_param': { 'field_name': 'plan_type', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class CarePlansListSecurity:
    drchrono_oauth2: shared_security.SchemeDrchronoOauth2 = dataclasses.field(metadata={'security': { 'scheme': True, 'type': 'oauth2' }})
    

@dataclass_json
@dataclasses.dataclass
class CarePlansList200ApplicationJSON:
    r"""CarePlansList200ApplicationJSON
    Paginated Result
    """
    
    data: Optional[list[shared_careplan.CarePlan]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    next: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('next') }})
    previous: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('previous') }})
    

@dataclasses.dataclass
class CarePlansListRequest:
    query_params: CarePlansListQueryParams = dataclasses.field()
    security: CarePlansListSecurity = dataclasses.field()
    

@dataclasses.dataclass
class CarePlansListResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    care_plans_list_200_application_json_object: Optional[CarePlansList200ApplicationJSON] = dataclasses.field(default=None)
    

File: SDKs/drchrono.com/v4 (Hunt Valley)/typescript/src/sdk/models/operations/careplanslist.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class CarePlansListQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=cursor" })
  cursor?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=doctor" })
  doctor?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=page_size" })
  pageSize?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=patient" })
  patient?: number;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=plan_type" })
  planType?: number;
}


export class CarePlansListSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  drchronoOauth2: shared.SchemeDrchronoOauth2;
}


// CarePlansList200ApplicationJson
/** 
 * Paginated Result
**/
export class CarePlansList200ApplicationJson extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data", elemType: shared.CarePlan })
  data?: shared.CarePlan[];

  @SpeakeasyMetadata({ data: "json, name=next" })
  next?: string;

  @SpeakeasyMetadata({ data: "json, name=previous" })
  previous?: string;
}


export class CarePlansListRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  queryParams: CarePlansListQueryParams;

  @SpeakeasyMetadata()
  security: CarePlansListSecurity;
}


export class CarePlansListResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  carePlansList200ApplicationJSONObject?: CarePlansList200ApplicationJson;
}
